Source/Scene/Vector3DTileContent.js

```diff
@@ -0,0 +1,580 @@
+            quantizedScale = Cartesian3.unpack(featureTableJson.QUANTIZED_VOLUME_SCALE);
+        }
+
+        var indices = new Uint32Array(arrayBuffer, byteOffset, indicesByteLength / sizeOfUint32);
+        byteOffset += indicesByteLength;
+
+        var positions;
+        var polylinePositions;
+        var pointsPositions;
+        if (isQuantized) {
+            positions = new Uint16Array(arrayBuffer, byteOffset, positionByteLength / sizeOfUint16);
+            byteOffset += positionByteLength;
+            polylinePositions = new Uint16Array(arrayBuffer, byteOffset, polylinePositionByteLength / sizeOfUint16);
+            byteOffset += polylinePositionByteLength;
+            pointsPositions = new Uint16Array(arrayBuffer, byteOffset, pointsPositionByteLength / sizeOfUint16);
+        } else {
+            positions = new Float32Array(arrayBuffer, byteOffset, positionByteLength / sizeOfFloat32);
+            byteOffset += positionByteLength;
+            polylinePositions = new Float32Array(arrayBuffer, byteOffset, polylinePositionByteLength / sizeOfFloat32);
+            byteOffset += polylinePositionByteLength;
+            pointsPositions = new Float32Array(arrayBuffer, byteOffset, pointsPositionByteLength / sizeOfFloat32);
+        }
+
+        byteOffset = featureTableBinary.byteOffset + featureTableJson.POLYGON_COUNT.byteOffset;
+        var counts = new Uint32Array(featureTableBinary.buffer, byteOffset, numberOfPolygons);
+
+        byteOffset = featureTableBinary.byteOffset + featureTableJson.POLYGON_INDEX_COUNT.byteOffset;
+        var indexCounts = new Uint32Array(featureTableBinary.buffer, byteOffset, numberOfPolygons);
+
+        byteOffset = featureTableBinary.byteOffset + featureTableJson.POLYLINE_COUNT.byteOffset;
+        var polylineCounts = new Uint32Array(featureTableBinary.buffer, byteOffset, numberOfPolylines);
+
+        var i;
+
+        if (numberOfPoints > 0) {
+            var decodeMatrix;
+            if (defined(quantizedOffset) && defined(quantizedScale)) {
+                decodeMatrix = Matrix4.fromTranslationRotationScale(new TranslationRotationScale(quantizedOffset, undefined, quantizedScale), new Matrix4());
+            } else {
+                decodeMatrix = Matrix4.IDENTITY;
+            }
+
+            this._billboardCollection = new BillboardCollection();
+            this._labelCollection = new LabelCollection();
+
+            for (i = 0; i < numberOfPoints; ++i) {
+                var x = pointsPositions[i * 3];
+                var y = pointsPositions[i * 3 + 1];
+                var z = pointsPositions[i * 3 + 2];
+                var position = Cartesian3.fromElements(x, y, z);
+                Matrix4.multiplyByPoint(decodeMatrix, position, position);
+                Cartesian3.add(position, center, position);
+
+                var b = this._billboardCollection.add();
+                b.position = position;
```

Comment:
Here and below, why not pass these to `add`?  To avoid the object literal?  I dunno, maybe that optimization is too soon given everything else this function does.